src/autoscaler/cf/service_plans.go

```diff
@@ -55,59 +56,45 @@ func (c *Client) getBrokerPlanGuid(ccServicePlanGuid string) (string, error) {
 	resp, err = c.httpClient.Do(req)
 
 	if err != nil {
-		logger.Error("do-request", err)
-		return "", fmt.Errorf("cf-client-get-broker-plan-guid: failed to execute request to CF API: %w", err)
+		c.logger.Error("do-request", err)
+		return nil, fmt.Errorf("cf-client-get-broker-plan-guid: failed to execute request to CF API: %w", err)
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
 		err = fmt.Errorf("cf-client-get-broker-plan-guid: failed to get service plan: %s [%d] %s", servicePlansUrl.String(), resp.StatusCode, resp.Status)
-		logger.Error("get-response", err)
-		return "", err
+		c.logger.Error("get-response", err)
+		return nil, err
 	}
 
 	result := &ServicePlanResource{}
 
 	err = json.NewDecoder(resp.Body).Decode(result)
 	if err != nil {
-		logger.Error("decode", err)
-		return "", fmt.Errorf("cf-client-get-broker-plan-guid: failed to decode response from CF API: %w", err)
+		c.logger.Error("decode", err)
+		return nil, fmt.Errorf("cf-client-get-broker-plan-guid: failed to decode response from CF API: %w", err)
 	}
-
-	brokerPlanGuid := result.Entity.UniqueId
-	logger.Info("found-guid", lager.Data{"brokerPlanGuid": brokerPlanGuid})
-	c.ccServicePlanToBrokerPlanGuid[ccServicePlanGuid] = brokerPlanGuid
-	c.brokerPlanGuidToCCServicePlanGuid[brokerPlanGuid] = ccServicePlanGuid
-
-	return brokerPlanGuid, nil
+	return result, nil
 }
 
+// GetServicePlan
+// This function does not really get a service plan but the service plan's broker catalog id
 func (c *Client) GetServicePlan(serviceInstanceGuid string) (string, error) {
```

Comment:
yeah ... we are going to do this in our next refactor where we move this logic into business side rather than the cf client. Which means that the tests for the business side will test this. Currently there are/were no tests for this code at all.